src/Microsoft.AspNetCore.SignalR.Client.Core/Internal/ConnectionLogScope.cs

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace Microsoft.AspNetCore.SignalR.Client.Internal
+{
+    internal class ConnectionLogScope : IReadOnlyList<KeyValuePair<string, object>>
+    {
+        private string _cachedToString;
+        private string _connectionId;
+
+        public string ConnectionId
+        {
+            get => _connectionId;
+            set
+            {
+                _cachedToString = null;
+                _connectionId = value;
+            }
+        }
+
+        public KeyValuePair<string, object> this[int index]
+        {
+            get
+            {
+                if (Count == 1 && index == 0)
+                {
+                    return new KeyValuePair<string, object>("ClientConnectionId", ConnectionId);
+                }
+
+                throw new ArgumentOutOfRangeException(nameof(index));
+            }
+        }
+
+        public int Count => string.IsNullOrEmpty(ConnectionId) ? 0 : 1;
+
+        public IEnumerator<KeyValuePair<string, object>> GetEnumerator()
+        {
+            for (var i = 0; i < Count; ++i)
+            {
+                yield return this[i];
+            }
+        }
+
+        IEnumerator IEnumerable.GetEnumerator()
+        {
+            return GetEnumerator();
+        }
+
+        public override string ToString()
+        {
+            if (_cachedToString == null)
+            {
+                if (!string.IsNullOrEmpty(ConnectionId))
+                {
+                    _cachedToString = string.Format(
+                        CultureInfo.InvariantCulture,
+                        "ClientConnectionId:{0}",
```

Comment:
Why not just ConnectionId

Reply:
ConnectionLogScope is copied and pasted from HttpConnection. ClientConnectionId is used there as well.

Reply:
Hrm, I now remember why we don't use `ConnectionId` on the server. It conflicts with the HTTP `ConnectionId` hosting puts in. The client is less likely to have that problem... unless you're using the client on an ASP.NET server. Maybe `ClientConnectionId` was better in the first place. So what I'm saying is... I'm flip-flopping ;)